tracing-opentelemetry/src/subscriber.rs

```diff
@@ -555,6 +569,25 @@ where
                     builder.status_code = Some(otel::StatusCode::Error);
                 }
 
+                if self.event_location {
+                    let builder_attrs = builder.attributes.get_or_insert(Vec::new());
+                    if let Some(file) = meta.file() {
+                        #[cfg(feature = "tracing-log")]
+                        builder_attrs.push(KeyValue::new("code.filepath", file.to_owned()));
+                        #[cfg(not(feature = "tracing-log"))]
+                        builder_attrs.push(KeyValue::new("code.filepath", Cow::Borrowed(file)));
+                    }
+                    if let Some(module) = meta.module_path() {
+                        #[cfg(feature = "tracing-log")]
+                        builder_attrs.push(KeyValue::new("code.namespace", module.to_owned()));
+                        #[cfg(not(feature = "tracing-log"))]
+                        builder_attrs.push(KeyValue::new("code.namespace", Cow::Borrowed(module)));
+                    }
+                    if let Some(line) = meta.line() {
+                        builder_attrs.push(KeyValue::new("code.lineno", line as i64));
+                    }
```

Comment:
hmm, this is kind of a shame; if `tracing-log` is enabled, _all_ events will get `to_owned` called on them, even the ones coming from `tracing` that always have `'static` metadata. I think it would be better to handle this at runtime as well. i think we can do that based on whether `normalized_meta` is `Some` (indicating that the event came from the `log` compatibility layer) or not --- if it is `Some`, we do need to convert the string slice into a string, because it won't be valid for the `&'static` lifetime. but, if `normalized_metadata` is `None`, then the event came from `tracing`, and its metadata contains `&'static` strings, so we can just use `event.metadata()`

i think this might look something like this:
```suggestion
                    #[cfg(feature = "tracing-log")]
                    let is_log = normalized_metadata.is_some();
                    #[cfg(not(feature = "tracing-log"))]
                    let is_log = false;
                    
                    if is_log {
                        if let Some(file) = meta.file() {
                            builder_attrs.push(KeyValue::new("code.filepath", file.to_owned()));
                        }
                        if let Some(module) = meta.module_path() {
                            builder_attrs.push(KeyValue::new("code.namespace", module.to_owned()));
                        }
                    } else {
                        if let Some(file) = event.metadata().file() {
                            builder_attrs.push(KeyValue::new("code.filepath", Cow::Borrowed(file)));
                        }
                        if let Some(module) = event.metadata().file() {
                            builder_attrs.push(KeyValue::new("code.namespace", Cow::Borrowed(module)));
                        }
                    }

                    if let Some(line) = meta.line() {
                        builder_attrs.push(KeyValue::new("code.lineno", line as i64));
                    }
```
(there's probably a less ugly way to write this, but this should work, i think)